### PR TITLE
Replace array_push calls behaving as $array[] since it works faster than invoking functions in PHP

### PR DESCRIPTION
**Pull request by lfluvisotto**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | http://php.net/manual/en/function.array-push.php > Note: If you use array_push() to add one element to the array, it's better to use $array[] = because in that way there is no overhead of calling a function. 
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | Run the store

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9207)

### CODE REVIEW

File: tools/htmlpurifier/HTMLPurifier.standalone.php [hunk 1]
```diff
@@ -13178,7 +13178,7 @@ public function validate($aIP, $config, $context)
             }
 
             while (count($first) < 8) {
-                array_push($first, '0');
+                $first[] = '0';
```
Comment (Quetzacoalt91) [inline tools/htmlpurifier/HTMLPurifier.standalone.php:13181]: Although I agree with the content of this PR, I'd prefer to leave this file untouched as we deal with an external lib.

**Review by mickaelandrieu — APPROVED** [review]

File: tools/htmlpurifier/HTMLPurifier.standalone.php (hunk 1 above)
Comment (mickaelandrieu) [reply tools/htmlpurifier/HTMLPurifier.standalone.php:13181]: As we use it directly, it's not an issue (doing this, it's like we have a fork).

Can we plan to use the composer [dependency](https://github.com/ezyang/htmlpurifier) instead?

Comment (lfluvisotto) [reply tools/htmlpurifier/HTMLPurifier.standalone.php:13181]: As HTML Purifier is a third party module it's better use via composer.

http://htmlpurifier.org/download#Composer

https://github.com/ezyang/htmlpurifier

Comment (mickaelandrieu) [reply tools/htmlpurifier/HTMLPurifier.standalone.php:13181]: Sure, but we need to plan it in another contribution (and check we didn't change anything in it, too :/ )

Comment (Quetzacoalt91) [reply tools/htmlpurifier/HTMLPurifier.standalone.php:13181]: It seems we already altered the lib, which is an issue :/

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks @lfluvisotto!

**Pull request merged**